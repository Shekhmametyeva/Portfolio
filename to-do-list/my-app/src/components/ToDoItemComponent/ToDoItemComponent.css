.todo__item {
    width: 100%;
    height: 50px;
    padding: 0 20px;
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 16px;
    background-color: #eee;
    
}
.todo__text {
    flex-grow: 1;
    text-align: center;
    font-size: 1.25rem;
    font-weight: 400;
}