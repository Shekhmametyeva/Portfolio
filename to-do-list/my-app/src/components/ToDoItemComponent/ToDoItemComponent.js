import './ToDoItemComponent.css';
import ButtonSvgComponent from '../ButtonSvgComponent/ButtonSvgComponent';
import FormItemComponent from '../FormItemComponent/FormItemComponent';



function ToDoItemComponent (props) {
    const input = props.inputItemOpen 
        ? <FormItemComponent 
                element={props.element.value} 
                rank={props.element.rank} 
                index={props.index} 
                editElement={props.editElement} 
                highlight={props.highlight} 
                closeInput={() => props.updateSetinputItem('')} 
                dataFull={props.dataFull} /> 
        : null;

    return (
        <div  
            data-testid='todo-item'
            className={props.element.complete 
                ?  (props.highlighted 
                    ? 'todo__item done highlighted' 
                    : 'todo__item done') 
                : (props.highlighted 
                    ? 'todo__item highlighted' 
                    : 'todo__item')} 
            onAnimationEnd={() => {props.highlight('')}} >
            <ButtonSvgComponent name={props.element.rank} class='todo__button' type='button'/>
            <h5 className='todo__text'>{props.element.value}</h5>
            <ButtonSvgComponent 
                callback={() => props.moveElAnotherDataset(props.index, props.element.value, props.element.rank)}
                name='complete' 
                class='todo__button todo__done'   
            />
            <ButtonSvgComponent data='edit' class='todo__button' name='edit' callback={props.updateSetinputItem} value={props.element.value} />
            <ButtonSvgComponent class='todo__button' name='delete' callback={props.deleteItem} value={props.index} />
            {input}
        </div>    
    )
}

export default ToDoItemComponent