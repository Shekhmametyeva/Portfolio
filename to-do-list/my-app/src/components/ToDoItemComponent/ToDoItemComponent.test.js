import {render, screen, fireEvent} from '@testing-library/react';
import ToDoItemComponent from './ToDoItemComponent';

describe('тестируем ToDoItemComponent', () => {
    test('render ToDoItemComponent', () => {
        render(<ToDoItemComponent element={{complete: false, rank: "Личные", value: "Alina"}} highlighted={false}/>);
        const item = screen.getByTestId('todo-item');
        expect(item).toBeInTheDocument(); 
        expect(item).toMatchSnapshot()
        const title = screen.getByText(/Alina/i);
        expect(title).toBeInTheDocument(); 
        const btns = screen.getAllByRole('button');
        expect(btns.length).toBe(4);
        
    })
    test('render ToDoItemComponent без классов', () => {
        render(<ToDoItemComponent element={{complete: false, rank: "Личные", value: "Alina"}} highlighted={false}/>);
        const item = screen.getByTestId('todo-item')
        expect(item).not.toHaveClass('highlighted', 'done');
    })
    test('render ToDoItemComponent с классом done', () => {
        render(<ToDoItemComponent element={{complete: true, rank: "Личные", value: "Alina"}} highlighted={false}/>);
        const item = screen.getByTestId('todo-item')
        expect(item).not.toHaveClass('highlighted');
        expect(item).toHaveClass('done');
    })
    test('render ToDoItemComponent с классом highlighted', () => {
        render(<ToDoItemComponent element={{complete: false, rank: "Личные", value: "Alina"}} highlighted={true}/>);
        const item = screen.getByTestId('todo-item')
        expect(item).not.toHaveClass('done');
        expect(item).toHaveClass('highlighted');
    })
    test('render ToDoItemComponent с классом done u highlighted', () => {
        render(<ToDoItemComponent element={{complete: true, rank: "Личные", value: "Alina"}} highlighted={true}/>);
        const item = screen.getByTestId('todo-item');
        expect(item).toHaveClass('highlighted', 'done');
    })
    test('при пропсе inputItemOpen появляется инпут', () => {
        render(<ToDoItemComponent element={{complete: true, rank: "Личные", value: "Alina"}} highlighted={true} inputItemOpen={true}/>);
        const input = screen.getByRole('textbox');
        expect(input).toBeInTheDocument();
    })
})