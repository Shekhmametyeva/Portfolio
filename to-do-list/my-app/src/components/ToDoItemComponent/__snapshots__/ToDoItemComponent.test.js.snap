// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тестируем ToDoItemComponent render ToDoItemComponent 1`] = `
<div
  class="todo__item"
  data-testid="todo-item"
>
  <button
    class="todo__button button"
    type="button"
  >
    <svg
      class="icon"
    >
      <use
        xlink:href="sprite.svg#Личные"
      />
    </svg>
  </button>
  <h5
    class="todo__text"
  >
    Alina
  </h5>
  <button
    class="todo__button todo__done button"
  >
    <svg
      class="icon"
    >
      <use
        xlink:href="sprite.svg#complete"
      />
    </svg>
  </button>
  <button
    class="todo__button button"
    data-testid="edit"
  >
    <svg
      class="icon"
    >
      <use
        xlink:href="sprite.svg#edit"
      />
    </svg>
  </button>
  <button
    class="todo__button button"
  >
    <svg
      class="icon"
    >
      <use
        xlink:href="sprite.svg#delete"
      />
    </svg>
  </button>
</div>
`;
